geospaas_processing/copiers.py

```diff
@@ -76,3 +78,18 @@ def copy(self):
             else:
                 LOGGER.debug("For dataset with id = %s, there is no local file address in the "
                              "database.", dataset.id)
+
+    def delete(self):
+        """
+        Delete the file(s) or symlink(s) after a certain period of obsoleteness (in days) of the
+        file(s) or symlink(s) inside the destination path. """
+        with os.scandir(self._destination_path) as scanned_dir:
+            for entry in scanned_dir:
+                if ('.snapshot' not in entry.path
+                and entry.stat(follow_symlinks=False).st_uid == os.getuid()
+                and time.time() - entry.stat(follow_symlinks=False).st_mtime >
+                    self.obsoleteness*24*3600):
+                    if (entry.is_file(follow_symlinks=False)):
+                        os.remove(entry.path)
+                    elif entry.is_symlink():
+                        os.unlink(entry.path)
```

Comment:
This code is equivalent to:
```python
if entry.is_file():
    os.remove(entry.path)
```

Reply:
In the case of having a healthy symlink, `os.remove` removes the original file! which is not what we want here in this code. We want that symlink which is in the spool folder is being deleted! not the original file. That is the reason why I differentiate between two case of `having symlink file in the spool folder` and `having a file in the spool folder`.

Reply:
`os.remove()` and `os.unlink()` behave the same regarding symlinks.

Reply:
Done in the next commit